monitoring/pkg/monitoring/exporter_contract_balance.go

```diff
@@ -0,0 +1,80 @@
+package monitoring
+
+import (
+	"context"
+	"math/big"
+	"sync"
+
+	commonMonitoring "github.com/smartcontractkit/chainlink-common/pkg/monitoring"
+)
+
+func NewNodeBalancesExporterFactory(log commonMonitoring.Logger, metrics Metrics) commonMonitoring.ExporterFactory {
+	return &nodeBalancesExporterFactory{
+		log,
+		metrics,
+	}
+}
+
+type nodeBalancesExporterFactory struct {
+	log     commonMonitoring.Logger
+	metrics Metrics
+}
+
+func (f *nodeBalancesExporterFactory) NewExporter(params commonMonitoring.ExporterParams) (commonMonitoring.Exporter, error) {
+	return &nodeBalancesExporter{
+		log:         f.log,
+		metrics:     f.metrics,
+		chainConfig: params.ChainConfig,
+		addrsSet:    []ContractAddressWithBalance{},
+	}, nil
+}
+
+type nodeBalancesExporter struct {
+	log         commonMonitoring.Logger
+	metrics     Metrics
+	chainConfig commonMonitoring.ChainConfig
+	addrsSet    []ContractAddressWithBalance
+	addrsMu     sync.Mutex
+}
+
+func (e *nodeBalancesExporter) Export(ctx context.Context, data interface{}) {
+	balanceEnvelope, isBalanceEnvelope := data.(BalanceEnvelope)
+	if !isBalanceEnvelope {
+		return
+	}
+
+	decimals := balanceEnvelope.Decimals
+	divisor := new(big.Int).Exp(new(big.Int).SetUint64(10), decimals, nil) // 10^(decimals)
+
+	for _, c := range balanceEnvelope.Contracts {
+		balanceAns := new(big.Int).Div(c.Balance, divisor)
+
+		e.metrics.SetBalance(
+			toFloat64(balanceAns),
+			c.Address.String(),
+			c.Name,
+			e.chainConfig.GetNetworkID(),
+			e.chainConfig.GetNetworkName(),
+			e.chainConfig.GetChainID())
+	}
+
+	e.addrsMu.Lock()
+	defer e.addrsMu.Unlock()
+
+	e.addrsSet = balanceEnvelope.Contracts
+
+}
+
+func (e *nodeBalancesExporter) Cleanup(_ context.Context) {
+	e.addrsMu.Lock()
+	defer e.addrsMu.Unlock()
+
+	for _, c := range e.addrsSet {
+		e.metrics.CleanupBalance(c.Address.String(), c.Name, e.chainConfig.GetNetworkID(), e.chainConfig.GetNetworkName(), e.chainConfig.GetChainID())
+	}
+}
+
+func toFloat64(bignum *big.Int) float64 {
+	val, _ := new(big.Float).SetInt(bignum).Float64()
+	return val
```

Comment:
ignoring the error? seems important if the number is too large to fit into a Float64

how unlikely is it to happen?

Reply:
the ignored value is not an error, but it just represents where the actual value was (up or down from the float64) because you lose precision from big.Float to float64.

but in general, it is very unllikely for contract balance to be too large to fit into float64 (	-1.7e+308 to +1.7e+308), especially since this is the number of STRK and not FRI (so we've already divided by 10^18)

Reply:
that's fair, and yeah the max is a very large number

my personal preference is to throw an error/log if something is too big / small to fit - but it also makes sense that it is highly unlikely